src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -189,11 +189,62 @@ void Reorder::createReorderPrimitive(const dnnl::memory::desc& srcDesc,
     if (!selectedPD)
         IE_THROW() << "Preferable primitive descriptor is not set.";
 
+    // How to query what data_types are supported by dnnl::reorder
+    auto isSupportedDataType = [](dnnl::memory::data_type data_type) {
+        switch (data_type) {
+        case dnnl::memory::data_type::f32:
+        case dnnl::memory::data_type::s32:
+        case dnnl::memory::data_type::bf16:
+        case dnnl::memory::data_type::s8:
+        case dnnl::memory::data_type::u8:
+        case dnnl::memory::data_type::f16:
+            return true;
+        default:
+            return false;
+        }
+        return false;
+    };
+
     const auto engine = getEngine();
     src_blocked = std::make_shared<Memory>(engine, DnnlExtensionUtils::makeDescriptor(srcDesc), srcPtr, false);
-
     dst_blocked = std::make_shared<Memory>(engine, DnnlExtensionUtils::makeDescriptor(dstDesc), dstPtr, false);
 
+    isPreConvert = !isSupportedDataType(srcDesc.get_data_type());
+    isPostConvert = !isSupportedDataType(dstDesc.get_data_type());
```

Comment:
First. This is not the best solution though. First of all, on this level we are working with oneDNN entities already. Thus it doesn't help us to avoid an exception during an attempt to create oneDNN memory descriptor, since on the cpu_plugin::Node level there might be an arbitrary precision, but  when we call `srcMemPtr->getDescWithType<DnnlMemoryDesc>()` on L179 we implicitly convert an arbitrary memory descriptor to an oneDNN memory desc, which is not guarantied to complete successfully in all the cases. Thus the check must be made even before the memory descriptor conversion.
Second. Using F32 as an intermediate type unconditionally  is not a good idea, since there might be rounding of integral values that may break accuracy. Thus. When both input and output types are not supported, we should simply reinterpret the smallest in size of the two types as any type of the same size supported by oneDNN and do the reordering with this type. Then reinterpret the type back and do the conversion.
Third. Since we have `Reorder::reorderData`, which is aimed to perform instant reordering on demand and it also have to support arbitrary precisions, I strongly encourage you to develop a Reorder Executor, similar to the existing DnnlExecutor class. For the intermediate buffer, the Scratchpad may be used when possible to reduce the amount of memory reallocations.

Reply:
OK, let's do：
1.  Precision check should be put before MemDesc conversion to oneDNN mem desc
2. Choose intermediate type: smallest size of input/output types, and the same size supported by oneDNN
3. Develop reorderExecutor class
4. Use Scratchpad for intermediate conversion buffer


For 2, we need know which type (same size)is supported, do we also use 'try create dnnl::reorder' to find out the support data type? Maybe a redefined list is better?

Reply:
> For 2, we need know which type (same size)is supported, do we also use 'try create dnnl::reorder' to find out the support data type? Maybe a redefined list is better?
Please refer to my previous comment https://github.com/openvinotoolkit/openvino/pull/19651#discussion_r1317393915. In short, to detect unsupported precisions we can reuse `DnnlExtensionUtils::IEPrecisionToDataType` with minor modifications. To detect unsuported precisions combinations we still need to try to instance a reorder primitve.